src/back/msl/writer.rs

```diff
@@ -1739,6 +1879,29 @@ impl<W: Write> Writer<W> {
         value: Handle<crate::Expression>,
         level: back::Level,
         context: &StatementContext,
+    ) -> BackendResult {
+        let comparisons = context.expression.index_info.choose_checks(
+            pointer,
+            context.expression.module,
+            context.expression.function,
+            context.expression.info,
+        );
+        if !comparisons.is_empty() {
+            write!(self.out, "{}if (", level)?;
+            self.put_condition(&comparisons, pointer, &context.expression)?;
+            writeln!(self.out, ")")?;
+            self.put_unchecked_store(pointer, value, level.next(), context)
+        } else {
+            self.put_unchecked_store(pointer, value, level, context)
```

Comment:
I've changed this to always put the store in brackets. That did make it awkward to move the `put_unchecked_store` out of the `if`. Since this was "nit", I didn't worry about it - but if you think it would still look better to hoist out that call (and add plumbing to indent it properly), we can totally do that.